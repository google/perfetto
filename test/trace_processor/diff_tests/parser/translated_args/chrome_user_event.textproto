# Chrome histogram hashes translation rules
packet {
  translation_table {
    chrome_user_event {
      action_hash_to_name { key: 10 value: "action1" }
      action_hash_to_name { key: 20 value: "action2" }
    }
  }
}

# Track for slice begin/end events.
packet {
  timestamp: 0
  trusted_packet_sequence_id: 1
  track_descriptor {
    uuid: 12345
    thread {
      pid: 123
      tid: 345
    }
    parent_uuid: 0
    chrome_thread {
      thread_type: 4
    }
  }
}

# Known action hash, should be translated to a name
packet {
  trusted_packet_sequence_id: 1
  timestamp: 1
  track_event {
    categories: "cat1"
    track_uuid: 12345
    type: 1
    name: "slice1"
    chrome_user_event {
      action_hash: 10
    }
  }
}
# Another known hash, should be translated to a name
packet {
  trusted_packet_sequence_id: 1
  timestamp: 2
  track_event {
    categories: "cat2"
    track_uuid: 12345
    type: 1
    name: "slice2"
    chrome_user_event {
      action_hash: 20
    }
  }
}
# Unknown hash, should not be translated to any name
packet {
  trusted_packet_sequence_id: 1
  timestamp: 3
  track_event {
    categories: "cat3"
    track_uuid: 12345
    type: 3
    name: "slice3"
    chrome_user_event {
      action_hash: 30
    }
  }
}

# Slice end events
packet {
  trusted_packet_sequence_id: 1
  timestamp: 6000
  track_event {
    track_uuid: 12345
    categories: "cat3"
    name: "slice3"
    type: 2
  }
}

packet {
  trusted_packet_sequence_id: 1
  timestamp: 6001
  track_event {
    track_uuid: 12345
    categories: "cat2"
    name: "slice2"
    type: 2
  }
}

packet {
  trusted_packet_sequence_id: 1
  timestamp: 6002
  track_event {
    track_uuid: 12345
    categories: "cat1"
    name: "slice1"
    type: 2
  }
}
