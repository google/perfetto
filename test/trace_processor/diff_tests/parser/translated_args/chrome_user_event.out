"flat_key","key","int_value","string_value"
"chrome_user_event.action","chrome_user_event.action","[NULL]","action1"
"chrome_user_event.action","chrome_user_event.action","[NULL]","action2"
"chrome_user_event.action_hash","chrome_user_event.action_hash",10,"[NULL]"
"chrome_user_event.action_hash","chrome_user_event.action_hash",20,"[NULL]"
"chrome_user_event.action_hash","chrome_user_event.action_hash",30,"[NULL]"
"event.category","event.category","[NULL]","cat1"
"event.category","event.category","[NULL]","cat2"
"event.category","event.category","[NULL]","cat3"
"event.name","event.name","[NULL]","slice1"
"event.name","event.name","[NULL]","slice2"
"event.name","event.name","[NULL]","slice3"
"utid","utid",1,"[NULL]"
